packages/dcos-integration-test/extra/test_packaging.py

```diff
@@ -0,0 +1,152 @@
+import logging
+
+import pytest
+
+log = logging.getLogger(__name__)
+
+
+def test_pkgpanda_api(dcos_api_session):
+
+    def get_and_validate_package_ids(path, node):
+        r = dcos_api_session.get(path, node=node)
+        assert r.status_code == 200
+        package_ids = r.json()
+        assert isinstance(package_ids, list)
+        for package_id in package_ids:
+            r = dcos_api_session.get(path + package_id, node=node)
+            assert r.status_code == 200
+            name, version = package_id.split('--')
+            assert r.json() == {'id': package_id, 'name': name, 'version': version}
+        return package_ids
+
+    active_buildinfo = dcos_api_session.get('/pkgpanda/active.buildinfo.full.json').json()
+    active_buildinfo_packages = sorted(
+        # Setup packages don't have a buildinfo.
+        (package_name, info['package_version'] if info else None)
+        for package_name, info in active_buildinfo.items()
+    )
+
+    def assert_packages_match_active_buildinfo(package_ids):
+        packages = sorted(map(lambda id_: tuple(id_.split('--')), package_ids))
+        assert len(packages) == len(active_buildinfo_packages)
+        for package, buildinfo_package in zip(packages, active_buildinfo_packages):
+            if buildinfo_package[1] is None:
+                # No buildinfo for this package, so we can only compare names.
+                assert package[0] == buildinfo_package[0]
+            else:
+                assert package == buildinfo_package
+
+    for node in dcos_api_session.masters + dcos_api_session.all_slaves:
+        package_ids = get_and_validate_package_ids('pkgpanda/repository/', node)
+        active_package_ids = get_and_validate_package_ids('pkgpanda/active/', node)
+
+        assert set(active_package_ids) <= set(package_ids)
+        assert_packages_match_active_buildinfo(active_package_ids)
+
+
+# There is no standardized way of getting package requirements from the package description json,
+# e.g. nodes could be called 'brokers' or 'nodes' or something else. This was created by looking at
+# https://github.com/mesosphere/universe/blob/version-3.x/repo/packages/K/kafka/39/config.json
+# 1.1.9-0.10.0.0
+KAFKA_PACKAGE_REQUIREMENTS = {
+    'number_of_nodes': 3,
+    'node': {
+        'disk': 5000,
+        'mem': 2304,
+        'cpus': 1
+    },
+    'executor': {
+        'disk': 0,
+        'mem': 256,
+        'cpus': 0.5
+    }
+}
+
+
+def get_cluster_resources(dcos_api_session):
```

Comment:
private? `_get_cluster_resources` ? 
Actually, given the single usage. My tendency is to inline it at the place it is used `dcos_api_session.get('mesos/state-summary').json()`